### PR TITLE
Only Build Clamor Once in the CI

### PR DESCRIPTION
**Pull request by aang114**
This saves time since we are currently building Clamor 2.5 times (2 `cargo build`s and 1 `cargo check`)

### CODE REVIEW

**Comment by aang114** [comment]
@sinkingsugar is it still too long?

**Comment by aang114** [comment]
> @sinkingsugar is it still too long?

There's only 2 ways I can make it faster now:
1. Running "Push to codecov.io" parallelly
2. combining "Check if Clamor Builds" and "Check Build for Benchmarking" into one step

Lemme know if this makes sense. Thanks

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -37,10 +37,13 @@ jobs:
       - uses: actions/setup-node@v3
         with:
           node-version: '14'  # The node-version input is optional. If not supplied, the node version from PATH will be used. However, it is recommended to always specify Node.js version and don't rely on the system one.
-
-      - name: Test RPCs
+          
+      - name: Check if Clamor Builds
+        run: |
+          RUSTFLAGS="-D missing-docs" cargo build
```
Comment (alebaffa) [inline .github/workflows/check.yml]: why `build` and not `check`?

**Comment by alebaffa** [comment]
IMHO - sorry to jump in - it makes sense to do both, yes.

**Comment by sinkingsugar** [comment]
> IMHO - sorry to jump in - it makes sense to do both, yes.

this

**Review by alebaffa — APPROVED** [review]

**Comment by sinkingsugar** [comment]
What's going on here?
Merge "Check if clamor build" with "Check if test pass" with push codecov... why we run ANY code when pushing to codecov. Build clamor with coverage ON from the beginning. Same for tests. So you run coverage at same time.
Push cover should not build and run anything.

**Review by sinkingsugar — APPROVED** [review]
GOod enough, let's stop wasting more time.

**Pull request merged**